lib/cog/command/pipeline/executor.ex

```diff
@@ -215,7 +232,38 @@ defmodule Cog.Command.Pipeline.Executor do
         fail_pipeline(state, :binding_error, "Error preparing to execute command pipeline '#{state.request["text"]}': #{msg}")
       bound ->
         {current_bound, bound_scope} = collect_bound(bound)
-        {:next_state, :get_options, %{state | current_bound: current_bound, scope: bound_scope}, 0}
+        {:next_state, :maybe_expand, %{state | current_bound: current_bound, scope: bound_scope}, 0}
+    end
+  end
+
+  @doc """
+  `expand` -> {:stop, :shutdown} | {:next_state, :get_options}
+
+  If a command begins with 'user' or 'site' it is treated as an alias and expanded.
+  """
+  def maybe_expand(:timeout, %__MODULE__{current: %{command: "user:" <> _user_alias}=current_bound}=state),
+    do: expand_alias(current_bound, state)
+  def maybe_expand(:timeout, %__MODULE__{current: %{command: "site:" <> _site_alias}=current_bound}=state),
+    do: expand_alias(current_bound, state)
+  def maybe_expand(:timeout, state),
+    do: {:next_state, :get_options, state, 0}
+
+  defp expand_alias(current_bound, state) do
+    case AliasExpander.expand(current_bound.command) do
+      {:ok, %Ast.Pipeline{invocations: invocations}} ->
+        cond do
+          state.current_depth + 1 > state.max_depth ->
+            Helpers.send_reply("Error expanding alias. I'm trying to can... but I just can't.", state.request, state.mq_conn)
+            fail_pipeline(state, :alias_expansion_error, "Error expanding alias. Too much recursion.")
+          true ->
+            [new_current|rest] = invocations
+            # We insert a marker here to determine how deep we have gone with expansion
+            rest = rest ++ [%Marker{alias_name: current_bound.command}]
```

Comment:
1. As I think I've said previously we can (and should) defer cycle detection until the executor has been refactored per #286. For now, a warning in Cog's docs about cyclical alias references should be sufficient.
1. Limiting the number of expansion passes used to expand a single alias is orthogonal to detecting alias reference cycles. The conceptual separation should be reflected in the code as well. If I'm reading the code correctly -- and I might not be -- it appears that we're co-mingling the two concepts via the marker.

Suggestions:

1. Remove alias cycle detection logic.
1. Add logic to `Cog.Command.AliasExpander` to discover when an alias expands to a pipeline with a command in the first position. It should accumulate intermediate pipelines, if any, it finds along the way while being careful to preserve pipeline order.
1. Once the expander can identify a successful alias expansion we can add a counter and limit the number of times it loops.